### PR TITLE
refactoring code

### PR DESCRIPTION
**Pull request by GuptaEesh**
>Refactoring code.
>Used useEffect api calls to specific pages where it is needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "react-icons": "^4.3.1",
     "react-router-dom": "^6.2.2",
     "react-scripts": "^5.0.0",
+    "react-toastify": "^8.2.0",
```
Comment (shivampandey0) [inline package.json]: remove unnecessary dependencies.

**Review by Ronak-Mutha — APPROVED** [review]
Code looks clean. Good variable naming and project structure. Good to merge.

**Review by shivampandey0 — APPROVED** [review]
Clean code with individual components for different sections of the page.
- Added a suggestion

**Review by FarhanMobashir — APPROVED** [review]
good refactor. 
good usage of helpers.

File: package.json (hunk 1 above)
Comment (GuptaEesh) [reply package.json]: Done, Thank you for pointing out such a minute flaw. 
This shows how keenly you have reviewed my code.

**Comment by GuptaEesh** [comment]
Got 3 review comments. Merging the branch now

**Pull request merged**